bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/overlay/FindReplaceOverlayAction.java

```diff
@@ -0,0 +1,50 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Vector Informatik GmbH and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.ui.internal.findandreplace.overlay;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.jface.bindings.keys.KeyStroke;
+
+class FindReplaceOverlayAction {
+	private final Runnable operation;
+
+	private final List<KeyStroke> shortcuts = new ArrayList<>();
+
+	FindReplaceOverlayAction(Runnable operation) {
+		this.operation = operation;
+	}
+
+	void addShortcuts(List<KeyStroke> shortcutsToAdd) {
+		this.shortcuts.addAll(shortcutsToAdd);
+	}
+
+	void execute() {
+		operation.run();
+	}
+
+	boolean executeIfMatching(KeyStroke keystroke) {
+		if (shortcuts.stream().anyMatch(keystroke::equals)) {
+			execute();
+			return true;
+		}
+		return false;
+	}
+
+	String addShortcutHintToTooltipText(String originalTooltipText) {
```

Comment:
This doesn't feel like a Responsibility of the `FindReplaceOverlayAction` to me (shortcutHints are closer to UI than to the functionality which this class is providing). In lack of a better idea: how about moving it to the utils?

Reply:
This class is (almost) only about UI. Shortcuts are a UI concept and this class attached them to an operation without adding any pure business logic. When moving this method to a utility class, you need to pass the shortcuts to that utility method, which would require this action to expose the shortcuts and require from every consumer to properly extract the shortcuts and pass them to that utility. That would render this class almost useless.

Reply:
The reasoning that shortcuts are UI concepts is very sound. When I read "Action", I instinctively thought of "something like a Runnable", which is why I commented.
Now that I undersatnd this better, I am totally fine with this method